src/nodedef.cpp

```diff
@@ -676,12 +676,12 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 
 	bool is_liquid = false;
 
-	u8 material_type = (alpha == 255) ?
-		TILE_MATERIAL_BASIC : TILE_MATERIAL_ALPHA;
```

Comment:
Yeah only water uses 'alpha' and not with a value of 255 so your code as it is seems fine.